### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by xrisaD**

### CODE REVIEW

File: gr_nlp_toolkit/processors/tokenizer.py [hunk 1]
```diff
@@ -26,95 +24,45 @@ def create_ids(text: str) -> List[int]:
...
             mask.append('0')
         else:
             # add sub-words to token
             tokenObj.subwords.append(t)
+            tokenObj._ids.append(i)
             mask.append('1')
+
+    # create text
+    for token in tokens:
+        token.text = create_text(token._ids)
```
Comment (NickSmyr) [inline gr_nlp_toolkit/processors/tokenizer.py]: Create setter and getter for roken

**Review by NickSmyr — CHANGES_REQUESTED** [review]

File: gr_nlp_toolkit/processors/tokenizer.py [hunk 2]
```diff
@@ -26,95 +24,45 @@ def create_ids(text: str) -> List[int]:
...
+def create_mask_and_tokens(input_tokens: List[str], input_ids: List[int]) -> Tuple[List[str], List[Token]]:
     mask = []
     tokens = []
+
     # for each token
-    for t in input_tokens:
+    for t, i in zip(input_tokens, input_ids):
         # it isn't a sub-word
         if not t.startswith("##"):
             #  create a token object
```
Comment (NickSmyr) [inline gr_nlp_toolkit/processors/tokenizer.py]: Create a mapping from subword indexes to word indexes so at the head annotations of the DependencyParsing processor we can refer to words and not subwords

**Review by NickSmyr — CHANGES_REQUESTED** [review]

File: gr_nlp_toolkit/processors/tokenizer.py (hunk 2 above)
Comment (xrisaD) [reply gr_nlp_toolkit/processors/tokenizer.py]: Done

File: gr_nlp_toolkit/processors/tokenizer.py (hunk 1 above)
Comment (xrisaD) [reply gr_nlp_toolkit/processors/tokenizer.py]: Done

**Review by NickSmyr — APPROVED** [review]

**Pull request merged**